core/unix/coredump.c

```diff
@@ -484,11 +484,17 @@ os_dump_core_internal(dcontext_t *dcontext)
     // of LOAD program header (number of sections minus one since shstrtab does not
```

Comment:
> Align the offset of section data with the page size. Without the padding, gdb does not retrieve the section data properly.

Same issue: this PR is not changing the data in individual sections.  This is all about the section header table.  Please make this clear as I was very confused until I looked at the actual code.